doc/user_guide/migrating.rst

```diff
@@ -0,0 +1,109 @@
+Migrating Legacy Projects
```

Comment:
I think this is and was the most important section to me. 
Thanks for adding that, @Nicoretti !